### PR TITLE
Add assertion that no span appears more than once

### PR DESCRIPTION
**Pull request by rchande**
**Customer scenario**

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

**Bugs this fixes:**

(either VSO or GitHub links)

**Workarounds, if any**

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

**Risk**

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

**Performance impact**

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

**Is this a regression from a previous update?**

**Root cause analysis:**

How did we miss it?  What tests are we adding to guard against it in the future?

**How was the bug found?**

(E.g. customer reported it vs. ad hoc testing)

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs [hunk 1]
```diff
@@ -223,7 +223,12 @@ private bool ShouldConsiderSymbol(ISymbol symbol)
                 list.Add(new DocumentHighlights(kvp.Key, spans.ToImmutableAndFree()));
             }
 
-            return list.ToImmutableAndFree();
+            var result = list.ToImmutableAndFree();
+
+            // Diagnosing flaky tests: assert that no span appears more than once
+            var groupedSpans = result.SelectMany(d => d.HighlightSpans).GroupBy(s => s);
+            Contract.ThrowIfTrue(groupedSpans.Any(s => s.Count() > 1), "Found duplicate spans!");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs]: Do not check this in :)  

If you want to assert at the test level, that's fine.  But this risks breaking things by adding a new contract that wasn't ever stated before.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs]: (is this just for tracking down a test failure?  if so, that' sfine.  but it should not go into the product as a release-check).

Comment (rchande) [inline src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs]: @CyrusNajmabadi Don't worry, I'm not planning on checking this in.

**Comment by rchande** [comment]
This is attempting to track down the sporadic reference highlighting tests like seen in https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/3340/

**Comment by rchande** [comment]
retest 
windows_release_vs-integration_prtest

**Comment by rchande** [comment]
retest windows_debug_vs-integration_prtest

**Comment by rchande** [comment]
retest windows_release_vs-integration_prtest

**Comment by rchande** [comment]
retest windows_release_vs-integration_prtest

**Pull request closed without merging**